### PR TITLE
090122 as ip blocklist

### PR DESCRIPTION
**Pull request by AcceleryntSecurityDev**
Change(s):
   - New playbook

   Reason for Change(s):
   - New playbook

   Testing Completed:
   - Yes

### CODE REVIEW

**Comment by v-spadarthi** [comment]
@devikamehra and @rushriva : Please review the playbooks and provide your comments. Thanks!!

**Comment by hollyollyoxenfree** [comment]
@devikamehra @rushriva @v-spadarthi : Hello all, I just wanted to follow up with this. Has anyone had a chance to review?

**Comment by v-atulyadav** [comment]
Hi @devikamehra and @rushriva : Can you please review the playbooks and provide your feedback accordingly?. Thanks!!

**Comment by v-atulyadav** [comment]
Hi @devikamehra and @rushriva : Can you please review the playbooks and provide your feedback accordingly?. Thanks!!

**Comment by AcceleryntSecurityDev** [comment]
@devikamehra @rushriva @v-spadarthi there was some missing documentation and code for using the Azure Key Vault for storing secrets. My most recent commits have fixed these issues.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has fixed some issues can you please review and provide feedback accordingly. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has fixed some issues can you please review and provide feedback accordingly. Thanks.

**Comment by AcceleryntSecurityDev** [comment]
@v-atulyadav @devikamehra @rushriva - I have updated the azuredeploy.json file per your comments about new requirements on my other PRs

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes can you please review and provide feedback accordingly. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes can you please review and provide feedback accordingly. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes 10 days ago can you please check those changes. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev made some changes ten days ago, so please check them. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , please check the changes made by @AcceleryntSecurityDev few days ago. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes can you please review and provide feedback accordingly. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev made a few changes almost 21 days ago, so please check them out

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes can you please review and provide feedback accordingly. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes can you please review.Thanks

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes can you please review and provide feedback accordingly. Thanks.

**Comment by v-spadarthi** [comment]
Hi @rushriva and @devikamehra , @AcceleryntSecurityDev has made some changes can you please review and provide feedback accordingly. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev made some changes twelve days ago, so please check them. Thanks.

**Comment by v-atulyadav** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes can you please review.Thanks

**Comment by v-spadarthi** [comment]
Hi @rushriva @devikamehra , @AcceleryntSecurityDev has made some changes can you please review.Thanks

**Review by manishkumar1991 — COMMENTED** [review]
@AcceleryntSecurityDev :

Endpoint call for fetching the named list ID's is failing  ,,, Kindly check and respond

image

**Comment by hollyollyoxenfree** [comment]
> @AcceleryntSecurityDev :
> 
> Endpoint call for fetching the named list ID's is failing ,,, Kindly check and respond
> 
> image

@manishkumar1991 I was unable to replicate. I tested the logic app again as well and all appears to be working. 
image

**Comment by v-atulyadav** [comment]
Hi @AcceleryntSecurityDev, could you please respond to the comment above?

**Comment by manishkumar1991** [comment]
> @v-atulyadav @manishkumar1991 I was unable to replicate Manish's issue. I tested the logic app again as well and all appears to be working. image

I will check on this again

**Comment by manishkumar1991** [comment]
@AcceleryntSecurityDev  ,

I have again gone through the readme steps you mentioned,

I have created the name location list 
 image

then i have followed the highlighted steps 
image

which is failing for me, 
Can you please double check if you forget to mention any details on readme file which needs to be done before doing this step

**Comment by v-atulyadav** [comment]
Hi @AcceleryntSecurityDev, please check and respond to Manish's comments. Thanks.

**Comment by AcceleryntSecurityDev** [comment]
@manishkumar1991
It may be that this request is failing through Microsoft Graph if you are signed in with a personal account rather than a work account. Could this be the issue? I haven't been able to find any other prerequisites that could be associated with your error.
https://learn.microsoft.com/en-us/graph/api/conditionalaccessroot-list-namedlocations?view=graph-rest-1.0&tabs=http
image

**Comment by manishkumar1991** [comment]
> @manishkumar1991 It may be that this request is failing through Microsoft Graph if you are signed in with a personal account rather than a work account. Could this be the issue? I haven't been able to find any other prerequisites that could be associated with your error. https://learn.microsoft.com/en-us/graph/api/conditionalaccessroot-list-namedlocations?view=graph-rest-1.0&tabs=http image

Will check internally and respond back

**Comment by v-atulyadav** [comment]
Hi @AcceleryntSecurityDev, Manish will reply soon regarding the same. Thanks

**Comment by v-atulyadav** [comment]
Hi @AcceleryntSecurityDev, due to permission issues, it is taking time to check, but we will update you as soon as possible. Thanks

File: Playbooks/AS-IP-Blocklist/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,373 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
+        "title": "AS-IP-Blocklist", 
+        "description": "This playbook is intended to be run from a Microsoft Sentinel Incident. It will add the IP address from Microsoft Sentinel Incidents to a Microssoft Azure Conditional Access Named Locations list, signifying compromised IP addresses.",
```
Comment (rahul0216) [inline Playbooks/AS-IP-Blocklist/azuredeploy.json:6]: Please correct spelling "Microssoft"

File: Playbooks/AS-IP-Blocklist/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+        "prerequisites": "1. A Microsoft Azure Named Locations list id is needed. 2. An Microsoft Azure App Registration is also needed for the Microwsoft Graph API. 3. A Microsoft Azure key vault containing the app registration secret must also be set up. Support for the set up and configuration of each of these items can be found here: https://github.com/Accelerynt-Security/AS-IP-Blocklist",
```
Comment (rahul0216) [inline Playbooks/AS-IP-Blocklist/azuredeploy.json:7]: 1.  "An Microsoft Azure App", I think it should be "A Microsoft Azure AD App"
2.  Please correct spelling "Microwsoft"

**Review by rahul0216 — COMMENTED** [review]
Mostly looks good. Please address couple of minor comments.

**Comment by rahul0216** [comment]
Successful configuration and test.

image

File: Playbooks/AS-IP-Blocklist/azuredeploy.json (hunk 2 above)
Comment (AcceleryntSecurityDev) [reply Playbooks/AS-IP-Blocklist/azuredeploy.json]: Updates have been made. Thank you.

**Review by rahul0216 — APPROVED** [review]
Changes looks good.

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**